homeassistant/components/versasense/sensor.py

```diff
@@ -11,12 +11,16 @@
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the sensor platform."""
     consumer = hass.data[DOMAIN]["consumer"]
-    peripheral = hass.data[DOMAIN][discovery_info["identifier"]]
-    parent_name = discovery_info["parent_name"]
-    unit = discovery_info["unit"]
-    measurement = discovery_info["measurement"]
 
-    async_add_entities([VSensor(peripheral, parent_name, unit, measurement, consumer)])
+    for entity_info in discovery_info:
+        parent_name = entity_info["parent_name"]
+        peripheral = hass.data[DOMAIN][entity_info["identifier"]]
+        unit = entity_info["unit"]
+        measurement = entity_info["measurement"]
+
+        async_add_entities(
```

Comment:
Please collect all entities first, then add all of them in a single all to `async_add_entities`.